File: CSharp/Theming/UIStyles/ThemeData.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Theming Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [ThemeData.cs]
 * 
 * 
 * This file is part of the Xceed DataGrid for WPF product. The use
 * and distribution of this Sample Code is subject to the terms
 * and conditions referring to "Sample Code" that are specified in
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 */
using System.Windows;
using Xceed.Wpf.DataGrid.Views;

namespace Xceed.Wpf.DataGrid.Samples.Theming.UIStyles
{
  public class ThemeData : DependencyObject
  {
    #region CONSTRUCTORS

    public ThemeData()
    {
    }

    #endregion CONSTRUCTORS

    #region ImageDataTemplate Property

    public static readonly DependencyProperty ImageDataTemplateProperty = DependencyProperty.Register( "ImageDataTemplate", typeof( DataTemplate ), typeof( ThemeData ) );

    public DataTemplate ImageDataTemplate
    {
      get
      {
        return ( DataTemplate )this.GetValue( ThemeData.ImageDataTemplateProperty );
      }
      set
      {
        this.SetValue( ThemeData.ImageDataTemplateProperty, value );
      }
    }

    #endregion ImageDataTemplate Property

    #region Description Property

    public static readonly DependencyProperty DescriptionProperty = DependencyProperty.Register( "Description", typeof( string ), typeof( ThemeData ) );

    public string Description
    {
      get
      {
        return ( string )this.GetValue( ThemeData.DescriptionProperty );
      }
      set
      {
        this.SetValue( ThemeData.DescriptionProperty, value );
      }
    }

    #endregion Description Property

    #region Theme Property

    public static readonly DependencyProperty ThemeProperty = DependencyProperty.Register( "Theme", typeof( Theme ), typeof( ThemeData ) );

    public Theme Theme
    {
      get
      {
        return ( Theme )this.GetValue( ThemeData.ThemeProperty );
      }
      set
      {
        this.SetValue( ThemeData.ThemeProperty, value );
      }
    }

    #endregion Theme Property

    #region ThemeGroup Property

    public static readonly DependencyProperty ThemeGroupProperty = DependencyProperty.Register( "ThemeGroup", typeof( string ), typeof( ThemeData ) );

    public string ThemeGroup
    {
      get
      {
        return ( string )this.GetValue( ThemeData.ThemeGroupProperty );
      }
      set
      {
        this.SetValue( ThemeData.ThemeGroupProperty, value );
      }
    }

    #endregion ThemeGroup Property

    #region IsTheme3D Property

    public static readonly DependencyProperty IsTheme3DProperty = DependencyProperty.Register( "IsTheme3D", typeof( bool ), typeof( ThemeData ) );

    public bool IsTheme3D
    {
      get
      {
        return ( bool )this.GetValue( ThemeData.IsTheme3DProperty );
      }
      set
      {
        this.SetValue( ThemeData.IsTheme3DProperty, value );
      }
    }

    #endregion IsTheme3D Property

    #region UseSystemTheme Property

    public static readonly DependencyProperty UseSystemThemeProperty = DependencyProperty.Register( "UseSystemTheme", typeof( bool ), typeof( ThemeData ) );

    public bool UseSystemTheme
    {
      get
      {
        return ( bool )this.GetValue( ThemeData.UseSystemThemeProperty );
      }
      set
      {
        this.SetValue( ThemeData.UseSystemThemeProperty, value );
      }
    }

    #endregion UseSystemTheme Property

    #region IsNewTheme Property

    public static readonly DependencyProperty IsNewThemeProperty = DependencyProperty.Register(
      "IsNewTheme",
      typeof( bool ),
      typeof( ThemeData ),
      new FrameworkPropertyMetadata( ( bool )false ) );

    public bool IsNewTheme
    {
      get
      {
        return ( bool )this.GetValue( ThemeData.IsNewThemeProperty );
      }
      set
      {
        this.SetValue( ThemeData.IsNewThemeProperty, value );
      }
    }

    #endregion

  }
}

File: Visual Basic .NET/Theming/UIStyles/ThemeData.vb
'
' * Xceed DataGrid for WPF - SAMPLE CODE - Theming Sample Application
' * Copyright (c) 2007-2024 Xceed Software Inc.
' * 
' * [ThemeData.vb]
' * 
' * 
' * This file is part of the Xceed DataGrid for WPF product. The use
' * and distribution of this Sample Code is subject to the terms
' * and conditions referring to "Sample Code" that are specified in
' * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
' 
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel
Imports System.Windows
Imports Xceed.Wpf.DataGrid.Views

Namespace Xceed.Wpf.DataGrid.Samples.Theming.UIStyles
  Public Class ThemeData
	  Inherits DependencyObject
	#Region "CONSTRUCTORS"

	Public Sub New()
	End Sub

	#End Region ' CONSTRUCTORS

	#Region "ImageDataTemplate Property"

	Public Shared ReadOnly ImageDataTemplateProperty As DependencyProperty = DependencyProperty.Register("ImageDataTemplate", GetType(DataTemplate), GetType(ThemeData))

	Public Property ImageDataTemplate() As DataTemplate
	  Get
		Return CType(Me.GetValue(ThemeData.ImageDataTemplateProperty), DataTemplate)
	  End Get
	  Set(ByVal value As DataTemplate)
		Me.SetValue(ThemeData.ImageDataTemplateProperty, value)
	  End Set
	End Property

	#End Region ' ImageDataTemplate Property

	#Region "Description Property"

	Public Shared ReadOnly DescriptionProperty As DependencyProperty = DependencyProperty.Register("Description", GetType(String), GetType(ThemeData))

	Public Property Description() As String
	  Get
		Return CStr(Me.GetValue(ThemeData.DescriptionProperty))
	  End Get
	  Set(ByVal value As String)
		Me.SetValue(ThemeData.DescriptionProperty, value)
	  End Set
	End Property

	#End Region ' Description Property

	#Region "Theme Property"

	Public Shared ReadOnly ThemeProperty As DependencyProperty = DependencyProperty.Register("Theme", GetType(Theme), GetType(ThemeData))

	Public Property Theme() As Theme
	  Get
		Return CType(Me.GetValue(ThemeData.ThemeProperty), Theme)
	  End Get
	  Set(ByVal value As Theme)
		Me.SetValue(ThemeData.ThemeProperty, value)
	  End Set
	End Property

	#End Region ' Theme Property

	#Region "ThemeGroup Property"

	Public Shared ReadOnly ThemeGroupProperty As DependencyProperty = DependencyProperty.Register("ThemeGroup", GetType(String), GetType(ThemeData))

	Public Property ThemeGroup() As String
	  Get
		Return CStr(Me.GetValue(ThemeData.ThemeGroupProperty))
	  End Get
	  Set(ByVal value As String)
		Me.SetValue(ThemeData.ThemeGroupProperty, value)
	  End Set
	End Property

	#End Region ' ThemeGroup Property

	#Region "IsTheme3D Property"

	Public Shared ReadOnly IsTheme3DProperty As DependencyProperty = DependencyProperty.Register("IsTheme3D", GetType(Boolean), GetType(ThemeData))

	Public Property IsTheme3D() As Boolean
	  Get
		Return CBool(Me.GetValue(ThemeData.IsTheme3DProperty))
	  End Get
	  Set(ByVal value As Boolean)
		Me.SetValue(ThemeData.IsTheme3DProperty, value)
	  End Set
	End Property

	#End Region ' IsTheme3D Property

	#Region "UseSystemTheme Property"

	Public Shared ReadOnly UseSystemThemeProperty As DependencyProperty = DependencyProperty.Register("UseSystemTheme", GetType(Boolean), GetType(ThemeData))

	Public Property UseSystemTheme() As Boolean
	  Get
		Return CBool(Me.GetValue(ThemeData.UseSystemThemeProperty))
	  End Get
	  Set(ByVal value As Boolean)
		Me.SetValue(ThemeData.UseSystemThemeProperty, value)
	  End Set
	End Property

	#End Region ' UseSystemTheme Property

	#Region "IsNewTheme Property"

	Public Shared ReadOnly IsNewThemeProperty As DependencyProperty = DependencyProperty.Register("IsNewTheme", GetType(Boolean), GetType(ThemeData), New FrameworkPropertyMetadata(CBool(False)))

	Public Property IsNewTheme() As Boolean
	  Get
		Return CBool(Me.GetValue(ThemeData.IsNewThemeProperty))
	  End Get
	  Set(ByVal value As Boolean)
		Me.SetValue(ThemeData.IsNewThemeProperty, value)
	  End Set
	End Property

	#End Region

  End Class
End Namespace